<?xml version="1.0"?>
<sdf version='1.6'>
<model name="soup_can">
  <link name="base_link_soup">

    <inertial>
      <pose frame=''>0 0 0 0 0 0</pose>
      <mass>0.349000</mass>
        <inertia>
          <ixx>0.000402</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.000200</iyy>
          <iyz>0</iyz>
          <izz>0.000402</izz>
        </inertia>
    </inertial>

    <visual name='base_link_soup'>
      <pose frame=''>0 0 0 0 0 0</pose>
      <geometry>
        <mesh>
          <uri>../meshes/soup_can_texture.obj</uri>
        </mesh>
      </geometry>
    </visual>

    <!--
      This places contact spheres on the corners of the visual cylinder and a
      *slightly* smaller inset contact cylinder (centered on the visual origin). This
      accounts for issues in the contact computation providing stable table
      contact *and* supports grasping.
      When the cylinder is in stable contact with the ground plane, the corner
      spheres will provide fixed contact points (simulating distributed contact
      points around the face).  However, for arbitrary grip configuration, the
      slightly inset cylinder will provide contact with a *slight* offset (in this
      case a deviation of 0.0005 m from the visual surface).
     -->
    <collision name='soup_can_collision'>
      <pose>0 0 0 -1.57 0 0</pose>
      <geometry>
        <cylinder>
          <radius>0.027850</radius>
          <length>0.095900</length>
        </cylinder>
      </geometry>
      <material>
        <diffuse>0 1 0 1.0</diffuse>
      </material>
    </collision>

    <collision name='point_collision1'>
      <pose>0.014425 0.048450 0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision2'>
      <pose>0.014425 -0.048450 0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision3'>
      <pose>-0.014425 0.048450 0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision4'>
      <pose>-0.014425 -0.048450 0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision5'>
      <pose>0.014425 0.048450 -0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision6'>
      <pose>0.014425 -0.048450 -0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision7'>
      <pose>-0.014425 0.048450 -0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

    <collision name='point_collision8'>
      <pose>-0.014425 -0.048450 -0.014425 0 0 0 </pose>
      <geometry>
        <sphere><radius>1e-7</radius></sphere>
      </geometry>
    </collision>

  </link>
</model>
</sdf>
